#
# Ubuntu: https://launchpad.net/bugs/295490
# Upstream: none
# Description: Don't hard code the /usr/lib/xulrunner-devel-1.9.0.x path
#  only available when the -dev package is installed. Get the libdir
#  dynamically at runtime by asking xulrunner.
#
---
 src/liferea.in |   18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

Index: liferea-1.4.18/src/liferea.in
===================================================================
--- liferea-1.4.18.orig/src/liferea.in
+++ liferea-1.4.18/src/liferea.in
@@ -9,27 +9,37 @@
 
 # autodetect alternate location when $0 provides information
 if [ "`basename $0`" != "$0" ]; then
        dist_bin=`dirname $0`
 fi
 
 params="$@"
 
+
 #
 # If we run with Gecko or XulRunner we need to set 
 # LD_LIBRARY_PATH (WebKit and GtkHTML do not need this).
 #
-if [ "@MOZILLA_LIB_ROOT@" != "" ]; then
+for xul_version in 1.9.1 1.9 ; do
+	if [ ! -x /usr/bin/xulrunner-$xul_version ] ; then
+		continue
+	fi
+	MOZILLA_LIB_ROOT=/usr/lib/xulrunner-`xulrunner-$xul_version --gre-version`
+	if [ ! -d $MOZILLA_LIB_ROOT ] ; then
+		echo "*warning* your xulrunner-$xul_version installation is broken"
+		continue
+	fi
 	if [ "$LD_LIBRARY_PATH" = "" ]; then
-		LD_LIBRARY_PATH=@MOZILLA_LIB_ROOT@
+		LD_LIBRARY_PATH=$MOZILLA_LIB_ROOT
 	else
-		LD_LIBRARY_PATH=@MOZILLA_LIB_ROOT@:$LD_LIBRARY_PATH
+		LD_LIBRARY_PATH=$MOZILLA_LIB_ROOT:$LD_LIBRARY_PATH
 	fi
 	export LD_LIBRARY_PATH
-fi
+	break
+done
 
 if [ -z "$DBUS_SESSION_BUS_ADDRESS" ]; then
 	eval `dbus-launch`
 	export DBUS_SESSION_BUS_ADDRESS
 fi
 
 exec $DEBUGGER "$dist_bin/liferea-bin" $params
